app/models/program.rb

```diff
@@ -1,6 +1,6 @@
 class Program < ActiveRecord::Base
-  belongs_to :flag
-  belongs_to :track
-  belongs_to :person
+  has_and_belongs_to_many :flags
+  has_and_belongs_to_many :tracks
```

Comment:
Declining to change because the link between programs and tracks is not rich; there is nothing to be gained by has_many :through.